apps/prairielearn/elements/pl-file-preview/pl-file-preview.js

```diff
@@ -110,7 +110,24 @@
               const type = blob.type;
               if (type === 'text/plain') {
                 const text = await blob.text();
-                code.textContent = text;
+                if (escapedFileName.endsWith('.ipynb')) {
+                  // importing the notebookjs library doesn't return an object, it sets the global variable 'ns'
+                  // importing DOMPurify sets the global variable DOMPurify.
+                  await Promise.all([
+                    import('marked'),
+                    import('dompurify'),
+                    import('notebookjs'),
+                  ]).then(async ([Marked]) => {
+                    nb.markdown = Marked.marked.parse;
+                    nb.sanitizer = DOMPurify.sanitize;
```

Comment:
For other elements that perform sanitization we also include the `SANITIZE_NAMED_PROPS` option, to avoid issues with conflicts with other elements.
```suggestion
                    nb.sanitizer = (code) => DOMPurify.sanitize(code, {SANITIZE_NAMED_PROPS: true});
```

Reply:
Did you push?

Reply:
No. I'm taking a few minutes to see if the showdown problem is a quick fix. (I should be working on getting my classes ready, but the temptation to play with code was too great :( ).

Reply:
> I should be working on getting my classes ready, but the temptation to play with code was too great :(

Preach...

Reply:
It turns out that the showdown problem is the same `this` vs. `this || globalThis` problem that `notebookjs` had. I'm going to prepare an MVE and PR for showdown, but then revert this PR to using Marked.  I'll make a new PR if/when showdown updates.  (I don't want to risk delaying this feature another three months.)  I'll let you know when I've pushed.

Reply:
For posterity, here's the PR into showdown: https://github.com/showdownjs/showdown/pull/1017

That project hasn't seen a merge in years, so I'm not optimistic about it being fixed 😢